samples/Storage/https-traffic-only/azurepolicy.json

```diff
@@ -12,10 +12,36 @@
                         "equals": "Microsoft.Storage/storageAccounts"
                     },
                     {
-                        "not": {
-                            "field": "Microsoft.Storage/storageAccounts/supportsHttpsTrafficOnly",
-                            "equals": "true"
-                        }
+                        "anyOf": [
+                            {
+                                "allOf": [
+                                    {
+                                        "value": "[requestContext().apiVersion]",
+                                        "in": [
+                                            "2018-11-01",
+                                            "2018-07-01",
+                                            "2018-03-01-preview",
+                                            "2018-02-01",
+                                            "2017-10-01",
+                                            "2017-06-01",
+                                            "2016-12-01",
+                                            "2016-05-01",
+                                            "2016-01-01",
+                                            "2015-06-15",
+                                            "2015-05-01-preview"
+                                        ]
+                                    },
+                                    {
+                                        "field": "Microsoft.Storage/storageAccounts/supportsHttpsTrafficOnly",
```

Comment:
this should still check "notEquals": "true" in the older api-versions, correct?  What is "exists": "false" checking?

Reply:
if you use notEquals true, the property would need to be there for the resource not to be non compliant.

On newer api versions, it doesn't matter if the property is not there because it is enabled by default, so equals false applies for both the new and old api versions (and that's needed so you don't leave a gap for new api versions forcing the value to false), but exists false only applies for old api versions.